core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -147,7 +150,10 @@ abstract class SparkFunSuite
     if (excluded.contains(testName)) {
       ignore(s"$testName (excluded)")(testBody)
     } else {
-      super.test(testName, testTags: _*)(testBody)
+      val timeout = sys.props.getOrElse("spark.test.timeout", "20").toLong
```

Comment:
Add a un-document config `spark.test.timeout` and default value is 20, Units is Minutes